src/app/pages/account/users/user-form/user-form.component.ts

```diff
@@ -435,4 +443,15 @@ export class UserFormComponent {
 
     return username.toLocaleLowerCase();
   }
+
+  private updateShellOptions(group: number, groups: number[]): void {
+    let ids: number[] = [];
+    ids = ids.concat(groups);
+    if (group && !ids.find((id) => id === group)) {
+      ids.push(group);
+    }
+    this.ws.call('user.shell_choices', [ids]).pipe(choicesToOptions(), untilDestroyed(this)).subscribe((options) => {
```

Comment:
It may be worth adding `take(1)` since every change in the `group` and `groups` fields invokes `updateShellOptions`, which creates a new observable until the form is closed. It seems that's not critical. Up to you.